lib/app/ctx/person.ex

```diff
@@ -3,24 +3,118 @@ defmodule App.Ctx.Person do
   import Ecto.Changeset
 
   schema "people" do
-    field :email, :binary
-    field :email_hash, :binary
-    field :familyName, :binary
-    field :givenName, :binary
-    field :key_id, :integer
-    field :password_hash, :binary
+    field :email, Fields.EmailEncrypted
+    field :email_hash, Fields.EmailHash
+    field :familyName, Fields.Encrypted
+    field :givenName, Fields.Encrypted
+    field :locale, :string
+    field :password, :string, virtual: true
+    field :password_hash, Fields.Password
+    field :picture, :binary
     field :username, :binary
-    field :username_hash, :binary
+    field :username_hash, Fields.Hash
     field :status, :id
     field :tag, :id
+    field :key_id, :integer
 
+    has_many :sessions, App.Ctx.Session, on_delete: :delete_all
     timestamps()
   end
 
   @doc false
   def changeset(person, attrs) do
     person
-    |> cast(attrs, [:username, :username_hash, :email, :email_hash, :givenName, :familyName, :password_hash, :key_id])
-    |> validate_required([:username, :username_hash, :email, :email_hash, :givenName, :familyName, :password_hash, :key_id])
+    |> cast(attrs, [:username, :email, :givenName, :familyName, :password_hash, :key_id, :locale, :picture])
+    |> validate_required([:username, :email, :givenName, :familyName, :password_hash, :key_id])
+    |> put_email_hash()
+  end
+
+  def google_changeset(profile, attrs) do
+    profile
+    |> cast(attrs, [:email, :givenName, :familyName, :picture, :locale])
+    |> validate_required([:email])
+    |> put_email_hash()
+    |> put_email_status_verified()
+  end
+
+  defp put_email_hash(changeset) do
+    case changeset do
+      %{valid?: true, changes: %{email: email}} ->
+        put_change(changeset, :email_hash, email)
+
+      _ ->
+        changeset
+    end
+  end
+
+  defp put_email_status_verified(changeset) do
+    status_verified = App.Ctx.get_status_verified()
+    case changeset do
+      %{valid?: true} ->
+        put_change(changeset, :status, status_verified.id)
+
+      _ ->
+        changeset
+    end
   end
+
+  @doc """
+  `transform_profile_data_to_person/1` transforms the profile data
+  received from invoking `ElixirAuthGoogle.get_user_profile/1`
+  into a `person` record that can be inserted into the people table.
+
+  ## Example
+
+    iex> transform_profile_data_to_person(%{
```

Comment:
need to update this example to use `:atoms` as keys in the Map.
Not necessary for _this_ PR. but good to do in next iteration. 💭